api/src/main/java/org/openmrs/api/UserService.java

```diff
@@ -301,16 +301,29 @@
 	/**
 	 * Changes the current user's password.
 	 * 
-	 * @param pw current password
+	 * @param secret current password or answer to secret question
 	 * @param pw2 new password
 	 * @throws APIException
 	 * @should match on correctly hashed sha1 stored password
 	 * @should match on incorrectly hashed sha1 stored password
 	 * @should match on sha512 hashed password
 	 * @should be able to update password multiple times
+	 * @should update password if secret is correct
+	 * @should not update password if secret is not correct
 	 */
 	@Logging(ignoredArgumentIndexes = { 0, 1 })
-	public void changePassword(String pw, String pw2) throws APIException;
+	public void changePassword(String secret, String pw2) throws APIException;
```

Comment:
You need to add a separate new method for this and keep the old one that takes the old and new password